[DOC-1269] [2026.1] OLTP sink for OTEL exports (#31897)
yugabyte/yugabyte-db: docs/content/stable/yugabyte-platform/alerts-monitoring/anywhere-export-configuration.md

| [AWS CloudWatch](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/WhatIsCloudWatch.html) | Database audit logs | |
| [Google Cloud Logging](https://cloud.google.com/logging/) | Database audit logs | |
| [Dynatrace](#dynatrace) | | Yes |
| [Loki](#loki) | Database audit logs | |

> | [OTLP](#otlp) | Database audit logs, PostgreSQL query logs | Yes |

Reviewer: PostgreSQL query logs -> we don't publicly support this yet. Best if we only mention audit logs and metrics. OTLP also supports metrics export

1. Enter your Dynatrace Access Token.
1. Click **Validate and Create Configuration**.

### Loki

[Grafana Loki](https://grafana.com/docs/loki/latest/) is a log aggregation system designed to store and query logs. YugabyteDB Anywhere can export database audit logs to a Loki-compatible endpoint (including self-hosted Loki and [Grafana Cloud](https://grafana.com/docs/grafana-cloud/send-data/logs/logs-with-loki/)).

#### Prerequisites

- Enable the Loki sink by setting the **Allow Loki Exporter in Telemetry Provider** Global Configuration option (config key `yb.telemetry.allow_loki`) to `true`.

    Refer to [Manage runtime configuration settings](../../administer-yugabyte-platform/manage-runtime-config/).

    The flag is enforced when you create or delete a Loki telemetry provider.

- A reachable Loki endpoint.

#### Create a Loki export configuration

To create an export configuration, do the following:

1. On the **Integrations** page, on the **Log & Metrics Export** tab, click **Add Configuration**.
1. Enter a name for the configuration.
1. Choose **Loki**.
1. Fill in the Loki-specific fields:
   - Enter the **Loki endpoint** URL. For example, `http://<loki-url>:<loki-port>`.

>    - **Organization / Tenant ID** (optional) — Organization ID is required when multi-tenancy is set up in Loki. Optional for Grafana Cloud as the authentication reroutes requests according to scope.

Reviewer 2: ```suggestion
   - **Organization / Tenant ID** (optional). Organization ID is required when multi-tenancy is set up in Loki. Optional for Grafana Cloud, as the authentication reroutes requests according to scope.
```

   - **Authentication Type**: **Basic Auth** or **No Auth**. If you choose **Basic Auth**, enter a **Username** and **Password**.
1. Click **Validate and Create Configuration**.

{{< note title="Loki configurations are immutable" >}}

After you create a configuration, you cannot edit it. To change settings, create a new configuration, reassign universes, and delete the old configuration.

{{< /note >}}

> After the configuration is created, attach it to a universe using the [database log export](../universe-logging/) workflow.

Reviewer 2: This is covered in the intro?

> For log export on Kubernetes universes, ensure the [OpenTelemetry Operator](https://opentelemetry.io/docs/platforms/kubernetes/operator/#getting-started) is installed on the cluster.

Reviewer 2: This is covered in intro?

### OTLP

YugabyteDB Anywhere supports [OTLP](https://opentelemetry.io/docs/) (OpenTelemetry Protocol) as a generic telemetry provider sink. An OTLP telemetry provider lets a universe stream database audit logs, and database metrics to any OTLP-compatible receiver using the standard OpenTelemetry wire format.

The OTLP sink is vendor-agnostic and works with any backend that speaks OTLP, including (but not limited to) [Cribl](https://cribl.io/), [Grafana Cloud](https://grafana.com/docs/grafana-cloud/), [New Relic](https://docs.newrelic.com/docs/opentelemetry/opentelemetry-introduction/), [Prometheus](https://prometheus.io/docs/guides/opentelemetry/) (3.0+), [VictoriaMetrics](https://docs.victoriametrics.com/guides/getting-started-with-opentelemetry/), and [Sumo Logic](https://help.sumologic.com/docs/send-data/opentelemetry-for-logs/).

> The same OTLP telemetry provider can be reused for [database audit logging](../universe-logging/), [database metrics export](../anywhere-metrics-export/), or both. OTLP uses the same OpenTelemetry Collector and universe export workflows as other telemetry providers.

Reviewer 2: ```suggestion
You can reuse the same OTLP telemetry provider for [database audit logging](../universe-logging/), [database metrics export](../anywhere-metrics-export/), or both. OTLP uses the same OpenTelemetry Collector and universe export workflows as other telemetry providers.
```

{{< note title="Enable OTLP before configuring" >}}

Set the **OTLP Exporter for Telemetry Provider** Global Configuration option (config key `yb.telemetry.allow_otlp`), which defaults to `false`. Refer to [Manage runtime configuration settings](../../administer-yugabyte-platform/manage-runtime-config/).

> - Note that till the flag is set to true, any REST API create and delete requests for OTLP telemetry providers return HTTP 400 with: `OTLP Exporter for Telemetry Provider is not enabled. Please set the runtime flag 'yb.telemetry.allow_otlp' to true.`

Reviewer 2: ```suggestion
If the flag is false, any REST API create and delete requests for OTLP telemetry providers return HTTP 400 with: `OTLP Exporter for Telemetry Provider is not enabled. Please set the runtime flag 'yb.telemetry.allow_otlp' to true.`
```

> The flag is enforced when you create or delete an OTLP telemetry provider.

Reviewer 2: What does this mean? Seems superfluous

Author: yes thats true. removing it.

1. Enter a name for the configuration.
1. Choose **Loki**.
1. Fill in the Loki-specific fields:
   - Enter the **Loki endpoint** URL. For example, `http://<loki-url>:<loki-port>`.
   - **Organization / Tenant ID** (optional) — Organization ID is required when multi-tenancy is set up in Loki. Optional for Grafana Cloud as the authentication reroutes requests according to scope.
   - **Authentication Type**: **Basic Auth** or **No Auth**. If you choose **Basic Auth**, enter a **Username** and **Password**.
1. Click **Validate and Create Configuration**.

{{< note title="Loki configurations are immutable" >}}

After you create a configuration, you cannot edit it. To change settings, create a new configuration, reassign universes, and delete the old configuration.

{{< /note >}}

After the configuration is created, attach it to a universe using the [database log export](../universe-logging/) workflow.

For log export on Kubernetes universes, ensure the [OpenTelemetry Operator](https://opentelemetry.io/docs/platforms/kubernetes/operator/#getting-started) is installed on the cluster.

### OTLP

> YugabyteDB Anywhere supports [OTLP](https://opentelemetry.io/docs/) (OpenTelemetry Protocol) as a generic telemetry provider sink. An OTLP telemetry provider lets a universe stream database audit logs, PostgreSQL query logs, and database metrics to any OTLP-compatible receiver using the standard OpenTelemetry wire format.

Reviewer: Let's leave out PostgreSQL query logs

Reviewer: Removed from the doc content draft as well.

> This is the _generic OTLP_ integration, not the dedicated [Dynatrace](#dynatrace) integration (Dynatrace also ingests via OTLP using vendor-specific configuration).

Reviewer: Do we need to mention the specifics of Dynatrace here?

The OTLP sink is vendor-agnostic and works with any backend that speaks OTLP, including (but not limited to) [Cribl](https://cribl.io/), [Grafana Cloud](https://grafana.com/docs/grafana-cloud/), [New Relic](https://docs.newrelic.com/docs/opentelemetry/opentelemetry-introduction/), [Prometheus](https://prometheus.io/docs/guides/opentelemetry/) (3.0+), [VictoriaMetrics](https://docs.victoriametrics.com/guides/getting-started-with-opentelemetry/), and [Sumo Logic](https://help.sumologic.com/docs/send-data/opentelemetry-for-logs/).

> The same OTLP telemetry provider can be reused for [database audit logging](../universe-logging/) and PostgreSQL query logs, [database metrics export](../anywhere-metrics-export/), or both. OTLP uses the same OpenTelemetry Collector and universe export workflows as other telemetry providers.

Reviewer: PGLE to be removed

The following `OTLPConfig` fields are accepted by the REST API but are not exposed in the UI:

| Field | Description |
| :---- | :---------- |
| `headers` | Map of extra HTTP headers attached to every export request. |
| `compression` | Exporter compression (`gzip`, `none`, `snappy`, `zstd`). UI create requests always send `gzip`. |
| `retryOnFailure` | Retry settings: `{ "initial_interval": "...", "max_interval": "...", "max_elapsed_time": "..." }` with duration strings such as `30s`, `1m`, `60m`. |

#### Additional configuration

| Runtime flag | Scope | Description |
| :----------- | :---- | :---------- |
| `yb.telemetry.skip_connectivity_validations` | Global | Skips connectivity and permission validations on create if your receiver is not reachable from YugabyteDB Anywhere at configuration time. |
| `yb.universe.otel_collector_max_memory` | Universe | Hard memory limit on the OpenTelemetry Collector process (kills the process if exceeded). To apply a change on an existing universe, re-run any OpenTelemetry configure API (configure metrics export, modify audit logging, or modify query logging). |

> #### OpenTelemetry Collector metrics

Reviewer: Was this section not added when DB metrics export support was being added? Best to leave it out in this instance as well if it was avoided in the previous effort
